### PR TITLE
applications: asset_tracker: Add watchdog to application.

### PR DESCRIPTION
**Pull request by jhn-nordic**
Adding watchdog functionality to asset tracker application.
Detects if system workqueue or the application workqueue stops executing
as intended.

Jira:TG91-156

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

### CODE REVIEW

**Comment by jhn-nordic** [comment]
Based on https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/applications/nrf_desktop/src/modules/watchdog.c

File: applications/asset_tracker/src/watchdog/watchdog.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+static void primary_feed_worker(struct k_work *work_desc)
+{
+	k_work_submit_to_queue(second_work_q, &wdt_data.second_workqueue_work);
+}
+static void secondary_feed_worker(struct k_work *work_desc)
+{
+	int err = wdt_feed(wdt_data.wdt_drv, wdt_data.wdt_channel_id);
+
+	LOG_DBG("Feeding watchdog");
+	if (err) {
```
Comment (jtguggedal) [inline applications/asset_tracker/src/watchdog/watchdog.c]: Missing newline before `if`

File: applications/asset_tracker/src/watchdog/watchdog.c [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	int err = -ENXIO;
+
+	data->wdt_drv = device_get_binding(DT_WDT_0_NAME);
+	if (data->wdt_drv == NULL) {
+		LOG_ERR("Cannot bind watchdog driver");
+		return err;
+	}
+
+	err = watchdog_timeout_install(data);
+
```
Comment (jtguggedal) [inline applications/asset_tracker/src/watchdog/watchdog.c]: Empty line can be removed

**Comment by rlubos** [comment]
@jhn-nordic A conflict showed up.

**Pull request merged**